go.mod

```diff
@@ -52,20 +52,20 @@ require (
 	gonum.org/v1/gonum v0.16.0
 	gotest.tools v2.2.0+incompatible
 	helm.sh/helm/v3 v3.17.3
-	k8s.io/api v0.32.5
-	k8s.io/apiextensions-apiserver v0.32.5
-	k8s.io/apimachinery v0.32.5
-	k8s.io/cli-runtime v0.32.5
-	k8s.io/client-go v0.32.5
+	k8s.io/api v0.33.1
+	k8s.io/apiextensions-apiserver v0.33.1
+	k8s.io/apimachinery v0.33.1
+	k8s.io/cli-runtime v0.33.1
+	k8s.io/client-go v0.33.1
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/kube-openapi v0.0.0-20250318190949-c8a335a9a2ff
-	k8s.io/kubectl v0.32.5
-	k8s.io/kubernetes v1.32.2
+	k8s.io/kubectl v0.33.1
+	k8s.io/kubernetes v1.33.1
 	k8s.io/utils v0.0.0-20250502105355-0f33e8f1c979
 	oras.land/oras-go/v2 v2.6.0
 	sigs.k8s.io/cli-utils v0.37.2
-	sigs.k8s.io/controller-runtime v0.20.4
-	sigs.k8s.io/controller-tools v0.17.3
+	sigs.k8s.io/controller-runtime v0.21.0
```

Comment:
A few words or warning on this:
* v0.21.0 [deprecates `Result.Requeue`](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.21.0), which we use extensively
* When quickly scanning changelogs between these 2 versions, I have not found anything suggesting that the implementation detail on which we rely (namely events with a higher timestamp being discarded from the queue if events with a lower timestamp are added for the same object) might have been altered, but this may need a closer look.
@0xavi0, do you know if our tests would detect such a change?

Also, [v0.20.0](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.20.0) introduces a priority queue :eyes:

Reply:
Yes, that would be a good idea; it could target 2.13.0 for instance.

Reply:
Resolving following offline discussion. We've concluded that, should the `controller-runtime` implementation detail on which we rely be undone, this would result in more reconcile cycles, but not break anything.
We're still keeping an eye on this though, and are planning to rewrite the `GitRepo` polling mechanism to use more explicit polling; see #3394.